### PR TITLE
timing profile breakdown: take into account event.cat

### PR DESCRIPTION
**Pull request by luluz66**
1. use cat = "remote output download" for reporting download
   outputs
2. use both name = "subprocess.run" and cat = "local action execution"
   for "Executing locally"
3. Remove Objc and Swift compilation. Events with "SwiftCompile" and
   "ObjcCompile" can be in two different categories "local action execution" and
   "Remote execution process wall time". As a result, we are double accounting
   for SwiftComiple and ObjcCompile

**Related issues**:
https://github.com/buildbuddy-io/buildbuddy-internal/issues/1831

### CODE REVIEW

File: app/invocation/invocation_timing_card.tsx [hunk 1]
```diff
@@ -140,6 +142,12 @@ export default class InvocationTimingCardComponent extends React.Component<Props
         } else {
           this.state.durationMap.set(event.name, event.dur);
         }
+
+        if (this.state.durationByCategoryMap.get(event.cat)) {
```
Comment (jdhollen) [inline app/invocation/invocation_timing_card.tsx:146]: i think if you wanna make this a one-liner, you can do...

`set(event.cat, (this.state.durationByCategoryMap.get(event.cat) || 0) + event.dur)`

side note, this usage of a map in state is.. unusual and the map might as well be a regular member of the class (updating it this way wouldn't cause a re-render), but that's unrelated to this change.

File: app/invocation/invocation_breakdown_card.tsx [hunk 2]
```diff
@@ -6,6 +6,7 @@ import { getChartColor } from "../util/color";
 
 interface Props {
   durationMap: Map<string, number>;
+  durationByCategoryMap: Map<string, number>;
```
Comment (jdhollen) [inline app/invocation/invocation_breakdown_card.tsx:9]: seeing as these are both Map<string,number> with very similar names, it's probably worth documenting what the difference is and what can be found in each one

**Review by jdhollen — CHANGES_REQUESTED** [review]

File: app/invocation/invocation_breakdown_card.tsx (hunk 2 above)
Comment (luluz66) [reply app/invocation/invocation_breakdown_card.tsx:11]: Done.

File: app/invocation/invocation_timing_card.tsx (hunk 1 above)
Comment (luluz66) [reply app/invocation/invocation_timing_card.tsx]: Done.

**Review by jdhollen — APPROVED** [review]

**Review by siggisim — APPROVED** [review]

**Pull request merged**